tests/countries/test_guinea.py

```diff
@@ -0,0 +1,162 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.guinea import Guinea, GN, GIN
+from tests.common import CommonCountryTests
+
+
+class TestGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(Guinea)
+
+    def test_country_aliases(self):
+        self.assertAliases(Guinea, GN, GIN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Guinea(years=1957))
+
+    def test_second_republic_day(self):
+        name = "Jour de la Deuxième République"
+        self.assertHolidayName(name, (f"{year}-04-03" for year in range(1958, 2021)))
+        self.assertNoHolidayName(name, Guinea(years=range(2022, 2050)))
+
+    def test_all_saints_day(self):
+        name = "La Toussaint"
+        self.assertHolidayName(name, (f"{year}-11-01" for year in range(1958, 2021)))
+        self.assertNoHolidayName(name, Guinea(years=range(2022, 2050)))
```

Comment:
`assertNoHolidayName` can be simplified a bit more here (remember to change other test case's start date to 1959 as well):
```suggestion
    def test_second_republic_day(self):
        name = "Jour de la Deuxième République"
        self.assertHolidayName(name, (f"{year}-04-03" for year in range(1959, 2021)))
        self.assertNoHolidayName(name, range(2022, 2050))

    def test_all_saints_day(self):
        name = "La Toussaint"
        self.assertHolidayName(name, (f"{year}-11-01" for year in range(1959, 2021)))
        self.assertNoHolidayName(name, range(2022, 2050))
```

Reply:
It would be more accurate to use `range(1959, 2022)`.

Reply:
Good catch, fixed 🙏